### ISSUE TITLE
Image block: Incorrect image width with aspect ratio in block editor

### ISSUE THREAD

**Issue by ajlende** [WordPress/gutenberg#52219]
### Description

In the post editor, if you have an image block with an image that is smaller than the width of the post container and you specify an aspect ratio without selecting a width or height dimension, the image is incorrectly scaling up to fill the width of the container.

### Step-by-step reproduction instructions

Screenshots use https://github.com/WordPress/wordpress-develop/blob/62b286f9b2ebc7bb609827a09fcf5136cac5fa01/tests/phpunit/data/images/a2-small.jpg for testing.

1. Use the following image block content or create a block includes an aspect ratio (and scale), but no width or height.

```html
<!-- wp:image {"id":443,"aspectRatio":"3/4","scale":"cover","sizeSlug":"full","linkDestination":"none"} -->
<figure class="wp-block-image size-full"><img src="http://wp.local/wp-content/uploads/2023/06/a2-small.jpg" alt="" class="wp-image-443" style="aspect-ratio:3/4;object-fit:cover"/></figure>
<!-- /wp:image -->
```

3. See that the image fills the width of the editor instead of the width of the image.

### Screenshots, screen recording, code snippet

![image-block-aspect-ratio-width-bug](https://github.com/WordPress/gutenberg/assets/5129775/6d99c148-998f-4762-9cd9-a2117c0f8a11)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by draganescu** [comment]
This does not happen if the image is picked from the media library 🤔

**Comment by Mamaduka** [comment]
Was this resolved via #52286?

**Comment by ajlende** [comment]
> Was this resolved via https://github.com/WordPress/gutenberg/pull/52286?

No, that PR exclusively changed the frontend, not the editor. I took a break from this last week, but I'm back on it now.

**Comment by ajlende** [comment]
This one is way more frustrating than it should be. The solution should be as easy as adding `fit-content` for `width` and `height` for the resizable container.

The problem is that `re-resizable` doesn't support `fit-content`. It looks like [the style is being overridden when they render the component](https://github.com/bokuweb/re-resizable/blob/da458ab823949ee1090316a3d50950ff89a04a7d/src/index.tsx#L322-L348). It overrides custom styles, passing in `fit-content` appends `px` resulting in `fit-contentpx`, `undefined` results in `'auto'`, and `null` crashes the component.

**Comment by stokesman** [comment]
I’ve found this to still be reproducible on trunk. I have #68666 open and it happens to fix this in my testing. I’d appreciate if anyone subscribed here has a chance to test this on that branch and report whether it fixes it for you.
